pinot-core/src/main/java/org/apache/pinot/core/segment/processing/timehandler/TimeHandler.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.processing.timehandler;
+
+import javax.annotation.Nullable;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * Handler for time values within the records:
+ * - Filter records based on the start/end time range
+ * - Round time values (for ROLLUP and DEDUP mode)
```

Comment:
Time handling can be combined with any merge type per the requirement, and they work independently. I can see potential use cases that requires dedupping the records with same dimensions but different time value (e.g. user always asks for distinct values).
Since TimeHandler is not associated with any merge type, modified the javadoc to not mention the merge type here.